tests/phpunit/api/v3/ACLPermissionTest.php

```diff
@@ -83,6 +83,22 @@ public function testContactGetNoResultsHook() {
   }
 
   /**
+   * Function tests that an empty where hook returns no results with edit my contact.
+   *
+   * CRM-16512 caused contacts with Edit my contact to be able to view all records.
+   */
+  public function testContactGetNoResultsHookWithEditMyContact() {
+    $this->createLoggedInUser();
+    $this->hookClass->setHook('civicrm_aclWhereClause', array($this, 'aclWhereHookNoResults'));
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'view my contact');
+    $result = $this->callAPISuccess('contact', 'get', array(
+      'check_permissions' => 1,
+      'return' => 'display_name',
+    ));
+    $this->assertEquals(0, $result['count']);
```

Comment:
It certainly confused me!

Ok why don't you update this test and we'll merge it in and I'll try to get it fixed over in the other PR.